hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -492,68 +489,92 @@ private void insertTokenBurn(RecordItem recordItem) {
+        token.setTotalSupply(tokenCreateTransactionBody.getInitialSupply());
+        token.setTreasuryAccountId(treasury);
+        token.setType(TokenTypeEnum.fromId(tokenCreateTransactionBody.getTokenTypeValue()));
 
-            if (tokenCreateTransactionBody.hasKycKey()) {
-                token.setKycKey(tokenCreateTransactionBody.getKycKey().toByteArray());
-            }
+        if (tokenCreateTransactionBody.hasFeeScheduleKey()) {
+            token.setFeeScheduleKey(tokenCreateTransactionBody.getFeeScheduleKey().toByteArray());
+        }
 
-            if (tokenCreateTransactionBody.hasSupplyKey()) {
-                token.setSupplyKey(tokenCreateTransactionBody.getSupplyKey().toByteArray());
-            }
+        if (tokenCreateTransactionBody.hasFreezeKey()) {
+            token.setFreezeKey(tokenCreateTransactionBody.getFreezeKey().toByteArray());
+        }
 
-            if (tokenCreateTransactionBody.hasWipeKey()) {
-                token.setWipeKey(tokenCreateTransactionBody.getWipeKey().toByteArray());
-            }
+        if (tokenCreateTransactionBody.hasKycKey()) {
+            token.setKycKey(tokenCreateTransactionBody.getKycKey().toByteArray());
+        }
 
-            Set<EntityId> autoAssociatedAccounts = insertCustomFees(tokenCreateTransactionBody.getCustomFeesList(),
-                    consensusTimestamp, true, tokenId);
-            autoAssociatedAccounts.add(treasury);
-            TokenFreezeStatusEnum freezeStatus = token.getFreezeKey() != null ? TokenFreezeStatusEnum.UNFROZEN :
-                    TokenFreezeStatusEnum.NOT_APPLICABLE;
-            TokenKycStatusEnum kycStatus = token.getKycKey() != null ? TokenKycStatusEnum.GRANTED :
-                    TokenKycStatusEnum.NOT_APPLICABLE;
-            autoAssociatedAccounts.forEach(account -> {
-                TokenAccount tokenAccount = new TokenAccount(tokenId, account);
-                tokenAccount.setAssociated(true);
-                tokenAccount.setCreatedTimestamp(consensusTimestamp);
-                tokenAccount.setFreezeStatus(freezeStatus);
-                tokenAccount.setKycStatus(kycStatus);
-                tokenAccount.setModifiedTimestamp(consensusTimestamp);
-
-                entityListener.onEntityId(account);
-                entityListener.onTokenAccount(tokenAccount);
-            });
+        if (tokenCreateTransactionBody.hasSupplyKey()) {
+            token.setSupplyKey(tokenCreateTransactionBody.getSupplyKey().toByteArray());
+        }
+
+        if (tokenCreateTransactionBody.hasWipeKey()) {
+            token.setWipeKey(tokenCreateTransactionBody.getWipeKey().toByteArray());
+        }
 
-            entityListener.onToken(token);
+        Set<EntityId> autoAssociatedAccounts = insertCustomFees(tokenCreateTransactionBody.getCustomFeesList(),
+                consensusTimestamp, true, tokenId);
+        autoAssociatedAccounts.add(treasury);
+        if (recordItem.getRecord().getAutomaticTokenAssociationsCount() > 0) {
```

Comment:
I'm not sure if there's a scenario where there's no auto associated on a token create per the main nodes but it's probably better to not differentiate between empty list and missing from protobuf. This is a more accurate check for pre-0.18 proto.

```suggestion
        if (recordItem.getRecord().transactionRecord.hasAutomaticTokenAssociations()) {
```

Reply:
it's a repeated field so it doesn't have a `has` method. If the field is missing, i.e., a pre-0.18 message, the default is an empty list created in the message constructor.